aws/resource_aws_ssm_patch_baseline.go

```diff
@@ -468,18 +482,31 @@ func expandAwsSsmPatchRuleGroup(d *schema.ResourceData) *ssm.PatchRuleGroup {
 		}
 
 		rule := &ssm.PatchRule{
-			ApproveAfterDays:  aws.Int64(int64(rCfg["approve_after_days"].(int))),
 			PatchFilterGroup:  filterGroup,
 			ComplianceLevel:   aws.String(rCfg["compliance_level"].(string)),
 			EnableNonSecurity: aws.Bool(rCfg["enable_non_security"].(bool)),
 		}
 
+		// Verify that at least one of approve_after_days or approve_until_date is set
+		approveAfterDays, _ := rCfg["approve_after_days"].(int)
+		approveUntilDate, _ := rCfg["approve_until_date"].(string)
+
+		if approveAfterDays > 0 && len(approveUntilDate) > 0 {
+			return nil, fmt.Errorf("Only one of approve_after_days or approve_until_date must be configured")
+		}
+
+		if len(approveUntilDate) > 0 {
+			rule.ApproveUntilDate = aws.String(approveUntilDate)
+		} else {
+			rule.ApproveAfterDays = aws.Int64(int64(approveAfterDays))
+		}
```

Comment:
Given that this apply-time error is no different than the API returning a similar error, we generally prefer to leave this to the API and simplify the resource logic. 👍 

```suggestion
		if v, ok := rCfg["approve_after_days"].(int); ok && v != 0 {
			rule.ApproveAfterDays = aws.Int64(int64(v))
		}

		if v, ok := rCfg["approve_until_date"].(string); ok && v != "" {
			rule.ApproveUntilDate = aws.String(v)
		}
```

Hopefully in the future the Terraform Plugin SDK can natively support `AtLeastOneOf` behaviors within `TypeList`/`TypeSet`.